coderd/httpmw/oauth2.go

```diff
@@ -56,6 +56,21 @@ func ExtractOAuth2(config OAuth2Config, client *http.Client) func(http.Handler)
 				return
 			}
 
+			// OIDC errors can be returned as query parameters. This can happen
+			// if for example we are providing and invalid scope.
+			// We should terminate the OIDC process if we encounter an error.
+			oidcError := r.URL.Query().Get("error")
+			errorDescription := r.URL.Query().Get("error_description")
```

Comment:
Per [the OIDC spec](https://openid.net/specs/openid-connect-core-1_0.html#rfc.section.3.1.2.6) only the `error` response parameter is required; the `error_description` field is optional so we should handle that being blank.

Also there is an `error_uri` parameter that might be present so we should check for that as well.

Reply:
I do handle an empty `error_description`. The `Details` part of the error is extra developer debug info. If it is empty, we provide no extra details, since we have no extra information.

I can add `error_uri` though :+1: